test/e2e/update_test.go

```diff
@@ -218,4 +218,74 @@ var _ = Describe("Podman update", func() {
 		Expect(session).Should(ExitCleanly())
 		Expect(session.OutputToString()).Should(ContainSubstring("500000"))
 	})
+
+	It("podman update persists changes", func() {
+		SkipIfCgroupV1("testing flags that only work in cgroup v2")
+		SkipIfRootless("many of these handlers are not enabled while rootless in CI")
+
+		testCtr := "test-ctr-name"
+		ctr1 := podmanTest.Podman([]string{"run", "-d", "--name", testCtr, "-m", "512m", ALPINE, "top"})
+		ctr1.WaitWithDefaultTimeout()
+		Expect(ctr1).Should(ExitCleanly())
+
+		inspect1 := podmanTest.Podman([]string{"inspect", "--format", "{{ .HostConfig.Memory }}", testCtr})
+		inspect1.WaitWithDefaultTimeout()
+		Expect(inspect1).Should(ExitCleanly())
+		Expect(inspect1.OutputToString()).To(Equal("536870912"))
+
+		exec1 := podmanTest.Podman([]string{"exec", testCtr, "cat", "/sys/fs/cgroup/memory.max"})
+		exec1.WaitWithDefaultTimeout()
+		Expect(exec1).Should(ExitCleanly())
+		Expect(exec1.OutputToString()).Should(ContainSubstring("536870912"))
```

Comment:
This should be moved into a helper function that just accepts the limit as arg, then you can deduplicate the 4 calls here. Also in the helper make sure to use `ExpectWithOffset(1, ...)...` to make the stack trace better.